backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -317,8 +321,17 @@ public FutureExperimentRunDAO(
                 .execute());
   }
 
-  private InternalFuture<Void> checkProjectPermission(
-      List<String> projId, ModelDBActionEnum.ModelDBServiceActions action) {
+  public interface CheckPermissionBasedOnResourceTypesFunction {
```

Comment:
Avoid interface inside class unless it _really_ belongs there. It makes the code harder to read. In this case, this interface seems generic.

Reply:
Created new package for interfaces and moved this interface to that package